builtin/aws/ecs/platform.go

```diff
@@ -910,6 +918,78 @@ func (p *Platform) resourceServiceDestroy(
 	return nil
 }
 
+func (p *Platform) resourceAlbRedirectListenerCreate(
+	ctx context.Context,
+	sg terminal.StepGroup,
+	sess *session.Session,
+	log hclog.Logger,
+	alb *Resource_Alb,
+	externalIngressPort ExternalIngressPort,
+
+	state *Resource_Alb_RedirectListener,
+) error {
+
+	s := sg.Add("Initiating ALB Redirect Listener creation")
+	defer s.Abort()
+
+	var HTTPListener *elbv2.Listener
+
+	if externalIngressPort == 443 {
+		elbsrv := elbv2.New(sess)
+
+		listeners, err := elbsrv.DescribeListenersWithContext(ctx, &elbv2.DescribeListenersInput{
```

Comment:
This means that we'll make two calls `DescribeListenersWithContext` in many cases - once when we create the normal listener, and once here. There isn't a great way to share this information between these two resources at this moment. Instead of creating a new resource for the redirect listener, can you add this logic to the existing listener resource?